test/mock_data/coral_cover.jl

```diff
@@ -0,0 +1,60 @@
+using ADRIA: YAXArray, DataCube, ZeroDataCube
+
+"""Raw model coral cover results, proportional to k-area."""
+function coral_cover(;
+    n_timesteps::Int64=10,
+    n_group_sizes::Int64=35,
+    n_locations::Int64=5,
+    n_scenarios::Int64=32
+)::YAXArray{Float64,4}
+    coral_cover = MockData.full_coral_cover(;
+        n_timesteps, n_group_sizes, n_locations, n_scenarios
+    )
+
+    location_weights::Array{Float64,4} = repeat(
+        rand(n_timesteps, n_locations, n_scenarios), 1, 1, 1, n_group_sizes
+    )
+
+    return coral_cover .* permutedims(location_weights, [1, 4, 2, 3])
+end
+
+"""Cover for each location at each timestep and scenario sums up to 1."""
+function full_coral_cover(
+    ;
+    n_timesteps::Int64=10,
+    n_group_sizes::Int64=35,
+    n_locations::Int64=5,
+    n_scenarios::Int64=32
+)::YAXArray{Float64,4}
+    cover_weights::Array{Float64,4} = rand(
+        n_timesteps,
+        n_group_sizes,
+        n_locations,
+        n_scenarios
+    )
+
+    # For each timestep and scenario the total cover of each location sums up to 1
+    cover_data::Array{Float64,4} = (cover_weights ./ sum(cover_weights; dims=2))
+
+    # This is to prevent the cover to grow above 1 due to floating point precision error
+    cover_data[:, end, :, :] .=
+        1 .- dropdims(
+            sum(cover_data[:, 1:(end - 1), :, :]; dims=2); dims=2
+        )
```

Comment:
Actually, could this be addressed [with `clamp!()`](https://docs.julialang.org/en/v1/base/math/#Base.Math.clamp!) ?

Reply:
I don't think so, because it's not a vector with values that should be 1 but are 0.9998 or 1.0002, but a vector with values that should sum up to 1 (with respect to one of the dimensions) but sometimes sum up to 0.9998 or 1.0002. That's why I'm taking the last entry of each col (dimension 2) and replacing by 1 minus the sum of the other elements in each col.

Reply:
Yeah, makes sense